cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -719,6 +722,17 @@ func (p *GitHubCiProvider) configurePipeline(
 // ensureGitHubLogin ensures the user is logged into the GitHub CLI. If not, it prompt the user
 // if they would like to log in and if so runs `gh auth login` interactively.
 func ensureGitHubLogin(ctx context.Context, ghCli github.GitHubCli, hostname string, console input.Console) (bool, error) {
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	if githubToken != "" {
+		console.Message(ctx,
+			fmt.Sprintf(
+				"%s\n%s",
+				output.WithWarningFormat("WARNING: A value was detected for the 'GITHUB_TOKEN' environment variable."),
+				"Having this variable set can interfere with GitHub integration if not properly configured.",
+			),
+		)
+	}
+
```

Comment:
You can remove this...

What we do in azd is to detect this env var and remove it from the env when running `gh cli` commands.

I am working on a fix to improve that approach,, as we are currenly only doing it when trying to set a secret.. but it should be right from the beggining... as `azd pipeline config` requires more elevated permissions than the ones granted to the GITHUB_TOKEN